WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -1,205 +1,676 @@
+
+    Initialize-WSLBlobNFS-Internal -Force:$Force
+    if($LastExitCode -ne 0)
+    {
         return
     }
 
-    # mountcommand and mountdrive are mandatory arguments for mountshare
-    if([string]::IsNullOrWhiteSpace($mountcommand))
+    Write-Success "WSL environment for WSLBlobNFS usage is initialized. Now, you can mount the SMB share using Mount-WSLBlobNFS."
+}
+
+function Mount-WSLBlobNFS
+{
+    <#
+    .SYNOPSIS
+        Mount Blob NFS share in Windows via WSL.
+
+    .DESCRIPTION
+        This command mounts the provided Blob NFS share in WSL, exports the share via SMB, and mounts the SMB share in Windows.
+
+    .EXAMPLE
+        PS> Mount-WSLBlobNFS -RemoteMount "account.blob.core.windows.net:account/container"
+        You can just provide the NFSv3 share address as below.
+
+    .EXAMPLE
+        PS> Mount-WSLBlobNFS -RemoteMount "mount -t nfs -o vers=3,proto=tcp account.blob.preprod.core.windows.net:/account/container /mnt/nfsv3share"
+        You can also provide the NFS mount command if you want to provide extra mount parameters.
+
+    .LINK
+        https://github.com/Azure/BlobNFS-wsl2
+
+    .NOTES
+        Author:  Azure Blob NFS
+        Website: https://github.com/Azure/BlobNFS-wsl2
+    #>
+
+    [CmdletBinding(PositionalBinding=$true)]
+    param(
+        # RemoteMount is your Blob NFS share address or the NFS mount command.
+        # If the RemoteMount is the NFS share address, then the share will be mounted with default mount parameters.
+        # If the RemoteMount is the NFS mount command, then the share will be mounted with the provided mount parameters.
+        # Check the examples for more details.
+        [Parameter(Mandatory = $true, Position=0)][string]$RemoteMount,
+
+        # MountDrive is the drive letter to mount the SMB share in Windows.
+        [Parameter(Position=1)][string]$MountDrive
+    )
+
+    # Set the verbosity preference for WSL based on the current preferences.
+    $verbosity = 0
+    if($VerbosePreference -eq "Continue")
+    {
+        $verbosity = 1
+    }
+    else
+    {
+        $verbosity = 0
+    }
+    Enable-Verbosity -verbosity $verbosity
+
+    Initialize-WSLBlobNFS-Internal
+    if($LastExitCode -ne 0)
     {
-        Write-Host "Mount command is not provided."
         return
     }
-    if([string]::IsNullOrWhiteSpace($mountdrive))
+
+    # Create a new MountDrive if not provided
+    if([string]::IsNullOrWhiteSpace($MountDrive))
+    {
+        # Get the first free drive letter
+        # 65 is the ASCII value of 'A'
+        (65..(65+25)).ForEach({
+            if ((-not (Get-PSDrive ([char]$_) -ErrorAction SilentlyContinue) -and [string]::IsNullOrWhiteSpace($MountDrive)))
+            {
+                $MountDrive = [char]$_ + ":"
+                Write-Output "Using $MountDrive to mount the share."
+            }})
+    }
+
+    # Check if the MountDrive is already in use or not.
+    $pathExists = Test-Path "$MountDrive"
```

Comment:
if above loop couldn't get a free drive $MountDrive will be null.
Check it after the foreach loop and log appropriate error "No free drive to use" or similar